src/mame/jaleco/megasys1.cpp

```diff
@@ -1962,32 +2139,33 @@ void megasys1_state::system_D(machine_config &config)
 ***************************************************************************/
 
 
-void megasys1_state::system_Z(machine_config &config)
+void megasys1_typez_state::system_Z(machine_config &config)
 {
 	/* basic machine hardware */
 	M68000(config, m_maincpu, SYS_A_CPU_CLOCK); /* 6MHz (12MHz / 2) */
-	m_maincpu->set_addrmap(AS_PROGRAM, &megasys1_state::megasys1Z_map);
-	TIMER(config, m_scantimer).configure_scanline(FUNC(megasys1_state::megasys1A_scanline), "screen", 0, 1);
+	m_maincpu->set_addrmap(AS_PROGRAM, &megasys1_typez_state::megasys1Z_map);
+	TIMER(config, m_scantimer).configure_scanline(FUNC(megasys1_typez_state::megasys1A_scanline), "screen", 0, 1);
 
 	Z80(config, m_audiocpu, 3000000); /* OSC 12MHz divided by 4 ??? */
-	m_audiocpu->set_addrmap(AS_PROGRAM, &megasys1_state::z80_sound_map);
-	m_audiocpu->set_addrmap(AS_IO, &megasys1_state::z80_sound_io_map);
+	m_audiocpu->set_addrmap(AS_PROGRAM, &megasys1_typez_state::z80_sound_map);
+	m_audiocpu->set_addrmap(AS_IO, &megasys1_typez_state::z80_sound_io_map);
 
 	/* video hardware */
 	SCREEN(config, m_screen, SCREEN_TYPE_RASTER);
 	m_screen->set_refresh_hz(56.18); // same as nmk16.cpp based on YT videos.
 	m_screen->set_vblank_time(ATTOSECONDS_IN_USEC(0));
 	m_screen->set_size(32*8, 32*8);
 	m_screen->set_visarea(0*8, 32*8-1, 2*8, 30*8-1);
-	m_screen->set_screen_update(FUNC(megasys1_state::screen_update));
+	m_screen->set_screen_update(FUNC(megasys1_typez_state::screen_update));
 	m_screen->set_palette(m_palette);
 
 	GFXDECODE(config, m_gfxdecode, m_palette, gfx_z);
 	PALETTE(config, m_palette).set_format(palette_device::RRRRGGGGBBBBRGBx, 0x800/2);
-	MCFG_VIDEO_START_OVERRIDE(megasys1_state,megasys1_z)
 
 	MEGASYS1_TILEMAP(config, m_tmap[0], m_palette, 256*0);
+	m_tmap[0]->set_screen_tag("screen");
 	MEGASYS1_TILEMAP(config, m_tmap[1], m_palette, 256*2);
+	m_tmap[1]->set_screen_tag("screen");
```

Comment:
You’ve got a device finder for the screen.  Why aren’t you using it in place of the literal tag here?